src/api/routes/address.ts

```diff
@@ -93,26 +95,26 @@ export function createAddressRouter(db: DataStore, chainId: ChainID): RouterWith
   });
 
   // get balances for STX, FTs, and counts for NFTs
-  router.getAsync('/:stx_address/balances', async (req, res) => {
+  router.getAsync('/:stx_address/balances', async (req, res, next) => {
     const stxAddress = req.params['stx_address'];
     if (!isValidPrincipal(stxAddress)) {
       return res.status(400).json({ error: `invalid STX address "${stxAddress}"` });
     }
 
+    const includeUnanchored = isUnanchoredRequest(req, res, next);
     const currentBlockHeight = await db.getCurrentBlockHeight();
     if (!currentBlockHeight.found) {
       return res.status(500).json({ error: `no current block` });
     }
 
+    const blockHeight = currentBlockHeight.result + (includeUnanchored ? 1 : 0);
+
     // Get balance info for STX token
-    const stxBalanceResult = await db.getStxBalanceAtBlock(stxAddress, currentBlockHeight.result);
-    const tokenOfferingLocked = await db.getTokenOfferingLocked(
-      stxAddress,
-      currentBlockHeight.result
-    );
+    const stxBalanceResult = await db.getStxBalanceAtBlock(stxAddress, blockHeight);
+    const tokenOfferingLocked = await db.getTokenOfferingLocked(stxAddress, blockHeight);
```

Comment:
I've trying to avoid an http request resulting in concurrent SQL queries. In this case, a Promise.all could make it harder to profile a potential sql bottleneck during a request.